rollup.config.js

```diff
@@ -2,57 +2,104 @@ import babel from 'rollup-plugin-babel'
 import replace from 'rollup-plugin-replace'
 import resolve from 'rollup-plugin-node-resolve'
 import commonjs from 'rollup-plugin-commonjs'
+import { terser } from 'rollup-plugin-terser'
 
-const NODE_ENV = process.env.NODE_ENV || 'development'
-const outputFile = fileName =>
-  NODE_ENV === 'production'
-    ? `./dist/prod/${fileName}.js`
-    : `./dist/dev/${fileName}.js`
+const ENV = process.env.NODE_ENV || 'development'
+const ENV_ALIASES = {
+  production: 'prod',
+  development: 'dev',
+  test: 'prod',
+}
+const ENTRY_FILES = ['factory', 'index', 'createElement']
 
-const plugins = () => [
-  replace({
-    'process.env.NODE_ENV': JSON.stringify(NODE_ENV),
-  }),
-  babel({
-    exclude: 'node_modules/**',
-  }),
-  resolve({
-    browser: true,
-  }),
-  commonjs(),
-]
+const plugins = () =>
+  [
+    replace({
+      'process.env.NODE_ENV': JSON.stringify(ENV),
+    }),
+    babel({
+      exclude: 'node_modules/**',
+    }),
+    resolve({
+      browser: true,
+    }),
+    commonjs(),
+    ENV !== 'development' ? terser() : null,
+  ].filter(Boolean)
 
-export default [
-  {
-    input: './src/factory.js',
+const confCreators = ENTRY_FILES.map(entryName => [
+  entryName,
+  format => ({
+    input: `./src/${entryName}.js`,
     output: {
-      file: outputFile('factory'),
-      format: 'umd',
-      name: 'styleSheetFactory',
+      format,
+      file: `dist/${entryName}.${format}.${ENV_ALIASES[ENV]}.js`,
+      compact: ENV === 'production',
     },
     plugins: plugins(),
+  }),
+])
+
+const EXTENRAL = {
+  createElement: ['react'],
+  index: ['inline-style-prefixer', 'fnv1a'],
+  factory: ['inline-style-prefixer', 'fnv1a'],
+}
+const CJS_CONFIG = confCreators.map(([entryName, creator]) => ({
+  ...creator('cjs'),
+  external: EXTENRAL[entryName],
+}))
+
+const ESM_CONFIG = confCreators.map(([entryName, creator]) => ({
```

Comment:
hey @Rich-Harris sorry to bother you, is this way of building libraries with Rollup ok? I am a Rollup-noob unfortunately

Reply:
definitely! if it works then it's 🆒